src/app/router/Welcome/Welcome.tsx

```diff
@@ -90,15 +24,74 @@ function WelcomeRouter() {
           hideProgressBar={true}
           className="w-fit max-w-2xl"
         />
-        <App />
+        <Routes>
+          <Route path="/" element={<Layout />}>
+            <Route index element={<SetPassword />} />
+            <Route path="choose-path">
+              <Route
+                index={true}
+                element={
+                  <ChooseConnectorPath
+                    title={i18n.t("translation:choose_path.title")}
+                    description={i18n.t("translation:choose_path.description")}
+                  />
+                }
+              ></Route>
+              <Route path="create" element={<AlbyWallet variant="create" />} />
+              <Route path="login" element={<AlbyWallet variant="login" />} />
+              <Route path="choose-connector">
+                <Route
+                  index={true}
+                  element={
+                    <ChooseConnector
+                      title={i18n.t("translation:choose_connector.title")}
+                      description={i18n.t(
+                        "translation:choose_connector.description"
+                      )}
+                      connectorRoutes={connectorRoutes}
+                    />
+                  }
+                ></Route>
+                {connectorRoutes.map((connectorRoute) => {
+                  if (connectorRoute.children) {
```

Comment:
Is this if check needed? 
Couldn't you just use `{connectorRoute.children?.map(...)}` and remove the else?

Reply:
It is an if/else condition.

Reply:
I mean, can't you replace the condition with just this?
```
                      <Route
                        key={connectorRoute.path}
                        path={connectorRoute.path}
                      >
                        <Route index element={connectorRoute.element} />
                        {connectorRoute.children?.map((connectorRoute) => (
                          <Route
                            key={connectorRoute.path}
                            path={connectorRoute.path}
                            element={connectorRoute.element}
                          />
                        ))}
                      </Route>
```

Reply:
Now I see. I don't know if this works. I took this from Options page. Now that in the kollider PR #2309 routing got changed, I will refactor this anyways once it's in master.

Reply:
I refactored to use the `renderRoutes` introduced by im-aditiya